sympy/ntheory/residue_ntheory.py

```diff
@@ -393,7 +393,7 @@ def _sqrt_mod_prime_power(a, p, k):
     if k == 1:
         if p == 2:
             return [ZZ(a)]
-        if not is_quad_residue(a, p):
+        if not (pow(a, (p - 1) // 2, p) == 1 or p < 3 or a % p < 2):
```

Comment:
`p < 3` means that `p == 2` that was already tested above.

Reply:
yes, i will change it.

Reply:
I would also consider taking the quick test `a % p < 2` first before the `pow()` condition.

Reply:
Done.